kernel/interrupt.rs

```diff
@@ -53,3 +57,10 @@ pub fn handle_irq(irq: u8) {
     interval_work();
     run_deferred_jobs();
 }
+
+pub fn init() {
+    let mut vectors = IRQ_VECTORS.lock();
+    for _ in 0..NUM_IRQ_NUMBERS {
+        vectors.push(IrqVector::new());
+    }
```

Comment:
That itself is going to cause at least 8 reallocations as you haven't reserved the size of the vector in the first place.